src/plugins/vis_type_xy/public/vis_component.tsx

```diff
@@ -0,0 +1,278 @@
+      series: XYChartSeriesIdentifier
+    ): ClickTriggerEvent | null => {
+      if (xAccessor !== null) {
+        return getFilterFromSeriesFn(visData)(series);
+      }
+
+      return null;
+    },
+    []
+  );
+
+  const handleFilterAction = useCallback(
+    (event: ClickTriggerEvent, negate = false) => {
+      props.fireEvent({
+        ...event,
+        data: {
+          ...event.data,
+          negate,
+        },
+      });
+    },
+    [props]
+  );
+
+  const canFilter = async (event: ClickTriggerEvent | null): Promise<boolean> => {
+    if (!event) {
+      return false;
+    }
+    const filters = await getDataActions().createFiltersFromValueClickAction(event.data);
+    return Boolean(filters.length);
+  };
+
+  const toggleLegend = useCallback(() => {
+    setShowLegend((value) => {
+      const newValue = !value;
+      props.uiState?.set('vis.legendOpen', newValue);
+      return newValue;
+    });
+  }, [props.uiState?.set]);
+
+  const setColor = useCallback(
+    (newColor: string, seriesLabel: string | number) => {
+      const colors = props.uiState?.get('vis.colors') || {};
+      if (colors[seriesLabel] === newColor) {
+        delete colors[seriesLabel];
+      } else {
+        colors[seriesLabel] = newColor;
+      }
+      props.uiState?.setSilent('vis.colors', null);
+      props.uiState?.set('vis.colors', colors);
+      props.uiState?.emit('colorChanged');
+    },
+    [props.uiState?.emit, props.uiState?.get, props.uiState?.set, props.uiState?.setSilent]
+  );
+
+  const { visData, visParams } = props;
+  const config = getConfig(visData, visParams);
+  const timeZone = getTimeZone();
+  const xDomain =
+    config.xAxis.scale.type === ScaleType.Ordinal ? undefined : getXDomain(config.aspects.x.params);
+  const hasBars = visParams.seriesParams.some(
+    ({ type, data: { id: paramId } }) =>
+      type === ChartType.Histogram && config.aspects.y.find(({ aggId }) => aggId === paramId)
+  );
+  const adjustedXDomain =
+    config.xAxis.scale.type === ScaleType.Ordinal
+      ? undefined
+      : getAdjustedDomain(visData.rows, config.aspects.x, timeZone, xDomain, hasBars);
+  const legendPosition = useMemo(() => config.legend.position ?? Position.Right, [
+    config.legend.position,
+  ]);
+  const isDarkMode = getThemeService().useDarkMode();
+  const getSeriesName = getSeriesNameFn(config.aspects, config.aspects.y.length > 1);
+
+  const getSeriesColor = useCallback(
+    (series: XYChartSeriesIdentifier) => {
+      const seriesName = getSeriesName(series);
+      if (!seriesName) {
+        return;
+      }
+
+      const overwriteColors: Record<string, string> = props.uiState?.get('vis.colors', {});
+
+      allSeries.push(seriesName);
+      return getColorsService().createColorLookupFunction(allSeries, overwriteColors)(seriesName);
+    },
+    [allSeries, getSeriesName, props.uiState?.get]
+  );
+
+  if (visParams.dimensions.splitRow || visParams.dimensions.splitRow) {
```

Comment:
Yeah this _**was**_ in both the renderer to prevent showing no results and here to display the warning.

I fixed this in 2c4d351 to only check in the renderer to display the callout above the chart and the no data UI for the chart.

![image](https://user-images.githubusercontent.com/19007109/98835553-30ad4e80-2406-11eb-8797-9c176020d792.png)